sys/parsoid.js

```diff
@@ -372,10 +372,10 @@ PSP.generateAndSave = function(hyper, req, format, currentContentRes) {
                 body: res.body[format].body
             };
             resp.headers.etag = mwUtil.makeETag(rp.revision, tid);
-            var responsePromise = self.wrapContentReq(hyper, req,
-                    P.resolve(resp), format, tid);
             return self.saveParsoidResult(hyper, req, format, tid, res)
             .then(function() {
+                var responsePromise = self.wrapContentReq(hyper, req,
+                        P.resolve(resp), format, tid);
```

Comment:
@Pchelolo, it's possible that this failure hid between all the content-type failures, and thus managed to sneak through. I restarted https://travis-ci.org/wikimedia/restbase/builds/119320343, which should show the failure fixed by this patch.